src/Nethermind/Nethermind.Core/StorageCell.cs

```diff
@@ -13,10 +13,13 @@ namespace Nethermind.Core
         public Address Address { get; }
         public UInt256 Index { get; }
 
-        public StorageCell(Address address, in UInt256 index)
+        public bool IsTransient { get; }
```

Comment:
Does not feel right. Shouldn't you use a different map/store to store transient value?

Reply:
it was like that before, but I don't see a problem with this approach
transient storage is same as storage, the only difference is that we don't store it in the database.
this flag gives us a good differentiation between transient/persistent storage and since this is also used to calculate the the HashCode, we can easily store both of them in the same dictionary

Reply:
Are you sure? This definitely feels weird to me. Are there like `TransientStorageCell` before? Or only `TransientStorage` which is addressed via `StorageCell`? I definitely don't think they should be in the same dictionary if possible.

Reply:
no TransientStorageCell. Another class that kept track of transient storage.
what is the issue with keeping them in same dictionary?

Reply:
Its implicit and can cause issues down the road. `transient` imply it could get completely removed/rollback later. If someone does not realize the dictionary can contain transient data, they might accidentally store them all or something similar. Its a weird and unexpected decision.

Reply:
okay, I will create another dictionary that stores the transient storage cells.

Reply:
i think I will think more about this and try to come up with better solution.
storing it separately creates duplicate code and structures doing exactly the same thing. then you have store two snapshots, each for persistent and transient storage.
adding one flag streamlines everything - you just have to and one check in CommitCore to omit it from pushing to the db.